### PR TITLE
Port fuerte changes from network pool branch

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

port over some changes to fuerte

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behaviour change can only be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as all shell http testS

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/6210/

### CODE REVIEW

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/4658/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/6237/

File: 3rdParty/fuerte/src/types.cpp [hunk 1]
```diff
@@ -195,55 +195,31 @@ std::string to_string(AuthenticationType type) {
...
-
-std::string to_string(ErrorCondition error) {
+std::string to_string(Error error) {
   switch (error) {
-    case ErrorCondition::NoError:
+    case Error::NoError:
       return "No Error";
-    case ErrorCondition::ErrorCastError:
+    case Error::ErrorCastError:
       return "Error: casting int to ErrorCondition";
```
Comment (danielhlarkin) [inline 3rdParty/fuerte/src/types.cpp]:
```suggestion
      return "Error: casting int to Error";
```

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/6300/

**Pull request merged**